In [10]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import chi2_contingency
import statsmodels.api as sm

In [11]:
from scipy.stats import mode

In [12]:
df = pd.read_csv("datos_whales.csv")

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 701 entries, 0 to 700
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Country      701 non-null    object 
 1   SeeWhales    701 non-null    int64  
 2   Age          691 non-null    float64
 3   Gender       701 non-null    object 
 4   Education    701 non-null    object 
 5   Income_AUD   674 non-null    object 
 6   Income       674 non-null    float64
 7   AdultWhales  701 non-null    int64  
 8   YoungWhales  701 non-null    object 
 9   Max_wtp      701 non-null    float64
 10  wtp          701 non-null    object 
 11  PayConserve  701 non-null    int64  
dtypes: float64(3), int64(3), object(6)
memory usage: 65.8+ KB


In [14]:
df = df[["SeeWhales","Education","Age","Income", "PayConserve", "Income_AUD"]]

In [15]:
df.describe()

SeeWhales         Age        Income  PayConserve
count  701.000000  691.000000    674.000000   701.000000
mean     0.126961   42.918958  44540.456973     0.660485
std      0.333168   15.562472  26362.209685     0.473883
min      0.000000   13.000000  10000.000000     0.000000
25%      0.000000   30.000000  25000.500000     0.000000
50%      0.000000   41.000000  35000.500000     1.000000
75%      0.000000   54.500000  65000.500000     1.000000
max      1.000000   82.000000  85000.500000     1.000000

In [16]:
df.mean()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  """Entry point for launching an IPython kernel.


SeeWhales          0.126961
Age               42.918958
Income         44540.456973
PayConserve        0.660485
dtype: float64

In [18]:
df["Age"] = df["Age"].fillna(42.918958)

In [19]:
df

SeeWhales             Education   Age   Income  PayConserve  \
0            1               Diploma  31.0  65000.5            1   
1            0                Degree  22.0  10000.0            0   
2            0                Degree  46.0  85000.5            0   
3            0               Diploma  37.0  25000.5            1   
4            0                Degree  66.0      NaN            0   
..         ...                   ...   ...      ...          ...   
696          0     Completed year 10  56.0  10000.0            1   
697          0     Completed year 10  38.0  35000.5            1   
698          0               Diploma  24.0  25000.5            1   
699          0  Post-graduate degree  50.0  35000.5            1   
700          0                Degree  32.0  85000.5            0   

            Income_AUD  
0    $60,001 - $70,000  
1    Less than $20,000  
2    More than $70,000  
3    $20,001 - $30,000  
4                  NaN  
..                 ...  
696  Less than $20,000  
697  $30,001 - $40,000  
698  $20,001 - $30,000  
699  $30,001 - $40,000  
700  More than $70,000  

[701 rows x 6 columns]

In [20]:
y = pd.crosstab(df["Education"], df["PayConserve"])

In [21]:
chi2_contingency(y)

(13.127920849377242,
 0.10752339615615682,
 8,
 array([[ 33.95149786,  66.04850214],
        [ 33.95149786,  66.04850214],
        [ 58.39657632, 113.60342368],
        [ 29.87731812,  58.12268188],
        [  0.33951498,   0.66048502],
        [ 28.85877318,  56.14122682],
        [  8.48787447,  16.51212553],
        [ 18.67332382,  36.32667618],
        [ 25.4636234 ,  49.5363766 ]]))

No hay relación entre educación y Pay Conserve, debido a que el valor P es mayor de 0.05

In [22]:
u = pd.crosstab(df["Income_AUD"], df["PayConserve"])

In [23]:
chi2_contingency(u)


(25.6142313740965,
 0.00026265667062334315,
 6,
 array([[28.91394659, 58.08605341],
        [38.55192878, 77.44807122],
        [20.60534125, 41.39465875],
        [26.58753709, 53.41246291],
        [16.9495549 , 34.0504451 ],
        [45.86350148, 92.13649852],
        [46.52818991, 93.47181009]]))

Si hay relación entre la variable Income_AUD y Pay Conserve, debido a que el valor P es menor de 0.05

In [24]:
ballena = pd.crosstab(df["SeeWhales"], df["PayConserve"])

In [25]:
chi2_contingency(ballena)


(5.418812987362997, 0.019920898752360208, 1, array([[207.7831669, 404.2168331],
        [ 30.2168331,  58.7831669]]))

Si hay relación entre la variable SeeWhales (Ver ballenas) y Pay Conserve, debido a que el valor P es menor de 0.05

In [26]:
edad = pd.crosstab(df["Age"], df["PayConserve"])

In [27]:
chi2_contingency(edad)

(63.89365523715995, 0.6186611121281225, 68, array([[ 1.01854494,  1.98145506],
        [ 0.67902996,  1.32097004],
        [ 1.35805991,  2.64194009],
        [ 1.69757489,  3.30242511],
        [ 1.01854494,  1.98145506],
        [ 1.01854494,  1.98145506],
        [ 1.35805991,  2.64194009],
        [ 0.33951498,  0.66048502],
        [ 3.05563481,  5.94436519],
        [ 3.05563481,  5.94436519],
        [ 7.46932953, 14.53067047],
        [ 8.82738944, 17.17261056],
        [ 8.48787447, 16.51212553],
        [ 7.46932953, 14.53067047],
        [ 2.37660485,  4.62339515],
        [ 4.07417974,  7.92582026],
        [ 3.05563481,  5.94436519],
        [ 4.7532097 ,  9.2467903 ],
        [ 4.41369472,  8.58630528],
        [ 6.79029957, 13.20970043],
        [ 3.05563481,  5.94436519],
        [ 5.09272468,  9.90727532],
        [ 5.77175464, 11.22824536],
        [ 4.41369472,  8.58630528],
        [ 5.77175464, 11.22824536],
        [ 5.43223966, 10.56776034],
        [ 3.39514979,

No hay relación entre la edad y Pay Conserve, debido a que el valor P es mayor de 0.05


In [28]:
df

SeeWhales             Education   Age   Income  PayConserve  \
0            1               Diploma  31.0  65000.5            1   
1            0                Degree  22.0  10000.0            0   
2            0                Degree  46.0  85000.5            0   
3            0               Diploma  37.0  25000.5            1   
4            0                Degree  66.0      NaN            0   
..         ...                   ...   ...      ...          ...   
696          0     Completed year 10  56.0  10000.0            1   
697          0     Completed year 10  38.0  35000.5            1   
698          0               Diploma  24.0  25000.5            1   
699          0  Post-graduate degree  50.0  35000.5            1   
700          0                Degree  32.0  85000.5            0   

            Income_AUD  
0    $60,001 - $70,000  
1    Less than $20,000  
2    More than $70,000  
3    $20,001 - $30,000  
4                  NaN  
..                 ...  
696  Less than $20,000  
697  $30,001 - $40,000  
698  $20,001 - $30,000  
699  $30,001 - $40,000  
700  More than $70,000  

[701 rows x 6 columns]

In [31]:
df["Income"] = df["Income"].fillna(44540.456973)

In [32]:
df

SeeWhales             Education   Age        Income  PayConserve  \
0            1               Diploma  31.0  65000.500000            1   
1            0                Degree  22.0  10000.000000            0   
2            0                Degree  46.0  85000.500000            0   
3            0               Diploma  37.0  25000.500000            1   
4            0                Degree  66.0  44540.456973            0   
..         ...                   ...   ...           ...          ...   
696          0     Completed year 10  56.0  10000.000000            1   
697          0     Completed year 10  38.0  35000.500000            1   
698          0               Diploma  24.0  25000.500000            1   
699          0  Post-graduate degree  50.0  35000.500000            1   
700          0                Degree  32.0  85000.500000            0   

            Income_AUD  
0    $60,001 - $70,000  
1    Less than $20,000  
2    More than $70,000  
3    $20,001 - $30,000  
4                  NaN  
..                 ...  
696  Less than $20,000  
697  $30,001 - $40,000  
698  $20,001 - $30,000  
699  $30,001 - $40,000  
700  More than $70,000  

[701 rows x 6 columns]

In [33]:
af = df.drop(columns=["Income_AUD", "Education"])

In [34]:
af

SeeWhales   Age        Income  PayConserve
0            1  31.0  65000.500000            1
1            0  22.0  10000.000000            0
2            0  46.0  85000.500000            0
3            0  37.0  25000.500000            1
4            0  66.0  44540.456973            0
..         ...   ...           ...          ...
696          0  56.0  10000.000000            1
697          0  38.0  35000.500000            1
698          0  24.0  25000.500000            1
699          0  50.0  35000.500000            1
700          0  32.0  85000.500000            0

[701 rows x 4 columns]

In [36]:
y = af["PayConserve"]
x = af[["SeeWhales", "Age","Income"]]
x = sm.add_constant(x)
model = sm.OLS(y,x)
reg = model.fit()
reg.summary()

/usr/local/lib/python3.7/dist-packages/statsmodels/tsa/tsatools.py:117: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:            PayConserve   R-squared:                       0.020
Model:                            OLS   Adj. R-squared:                  0.016
Method:                 Least Squares   F-statistic:                     4.792
Date:                Thu, 19 May 2022   Prob (F-statistic):            0.00260
Time:                        22:32:53   Log-Likelihood:                -463.52
No. Observations:                 701   AIC:                             935.0
Df Residuals:                     697   BIC:                             953.2
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.8048      0.063     12.717      0.000       0.681       0.929
SeeWhales      0.1502      0.054      2.794      0.005       0.045       0.256
Age           -0.0033      0.001     -2.849      0.005      -0.006      -0.001
Income     -4.765e-07   6.91e-07     -0.690      0.490   -1.83e-06    8.79e-07
==============================================================================
Omnibus:                     5925.880   Durbin-Watson:                   1.897
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              114.445
Skew:                          -0.656   Prob(JB):                     1.41e-25
Kurtosis:                       1.517   Cond. No.                     1.84e+05
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.84e+05. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

See Whales (Ver ballenas) incrementa en un 15% la probabilidad en la disposición a pagar dos dolares para proteger el habitat
La edad disminuye la probabilidad en un 0,3% la probabilidad en la disposición a pagar dos dolares para proteger el habitat
El ingrese disminuye la probabilidad en la disposición a pagar dos dolares para proteger el habitat
Las varaibles significativas son: See Whales (Ver ballenas), edad, el ingreso, y las que no son significativas: el ingreso